File: recursion/java/PrintFunction.java
public class PrintFunction {

    public void printNumber(int n){
        if (n <= 0){
            return;
        }
        System.out.println("value of n: "+n);
        printNumber(n-1);
    }

    public static void main(String[] args) {
        PrintFunction pf = new PrintFunction();
        pf.printNumber(5);
    }
}

File: recursion/cpp/PrintFunction.cpp
#include <bits/stdc++.h>
using namespace std;

void printNumber (int n){
    if (n <= 0){
        return;
    }
    cout << "value of n: " << n << "\n";
    printNumber(n-1);
}

int main()
{
    printNumber(5);
    return 0;
}